### PR TITLE
Allow to skip PostgreSQL backend dependent tests at runtime

### PR DESCRIPTION
**Pull request by strk**
Allows users to define a `QGIS_PGTEST_DB_SKIP` environment variable to skip tests requiring the test postgresql database to be run.

This is a spin-off from #51891 in response to @nyalldawson request expressed in https://github.com/qgis/QGIS/pull/51891#discussion_r1109132667

### CODE REVIEW

**Comment by strk** [comment]
CI is failing due to a bug in the workflow itself, see https://lists.osgeo.org/pipermail/qgis-developer/2023-February/065542.html

**Comment by strk** [comment]
I've filed #51917 as a spinoff of this one

File: .github/workflows/run-tests.yml
Comment (nyalldawson) [inline .github/workflows/run-tests.yml:411]: can you back out this change please?

Comment (strk) [reply .github/workflows/run-tests.yml:411]: I can, but it was suggested by the "clang-tidy" test author to do it beause as reported in #51917 the test is broken and would then make CI red and this PR unmergeable. See https://lists.osgeo.org/pipermail/qgis-developer/2023-February/065543.html

Comment (strk) [reply .github/workflows/run-tests.yml:411]: \cc @troopa81

Comment (lnicola) [reply .github/workflows/run-tests.yml:411]: Depending on how it's configured, it might still be possible to merge PRs with failing tests. Given the amount of issues with the test suite, I suspect it is.

Comment (strk) [reply .github/workflows/run-tests.yml:411]: I've created a spinoff of this PR as #51935 to just contain this commit. I would wait for that to be merged before updating this PR as updating this PR would have no other effect than triggering a new CI run that's known to fail ( as you can see already in the history of this PR: https://github.com/qgis/QGIS/actions/runs/4203324613/jobs/7294511888 )

Comment (strk) [reply .github/workflows/run-tests.yml:411]: > Depending on how it's configured, it might still be possible to merge PRs with failing tests. Given the amount of issues with the test suite, I suspect it is.

Possibly, but I've heard more than once that developers don't even consider reviewing PRs until they have a green CI icon near them, and indeed this PR only got a review AFTER I pushed the commit disabling the broken test, which I did AFTER asking for help on the mailing list and receiving the suggestion to do the disabling here in order to "get the PR merged".

Comment (strk) [reply .github/workflows/run-tests.yml]: Given the observation you made about wanting to keep the clang-tidy test I've removed that commit from this PR. Can I get your approval on it now ? The CI run will fail, and I'm not even sure why github is running it again (maybe because master branch changed from the last time the code was checked)

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**
